File: reference/vector_samples_19_4_10/ASAM_XIL_API/ASAM_XIL_Client/Test/TestImplementation.cs
using Vector.CANoe.TFS;
using Vector.CANoe.Threading;

[TestClass]
public class MyTestClass
{
  
  [Export]
  [TestCase]
  public static void SimpleTestCaseNet()
  {
    var value = 5.5;
    var expectedValue = value + 1.0;
    
    Report.TestStep($"Setting TestVariableWriteable to {value}");
    XILServer.MAPort.TestVariableWriteable.Value = value;

    Execution.Wait(200);
    
    Report.TestStep($"Checking TestVariableReadable value. Expected: {expectedValue}");
    if(XILServer.MAPort.TestVariableReadable.Value == expectedValue)
    {
      Report.TestStepPass("Test passed!");
      return;
    }

    Report.TestStep($"Actual: {XILServer.MAPort.TestVariableReadable.Value}");
    Report.TestStepFail("Test failed!");
  }
}

File: reference/vector_samples_19_4_10/ASAM_XIL_API/ASAM_XIL_Client/Test/TestImplementation.py
import vector.canoe
import vector.canoe.threading
import vector.canoe.tfs
from application_layer import XILServer

@vector.canoe.tfs.export
@vector.canoe.tfs.test_case
def SimpleTestCasePy():
    value = 6.5
    expectedValue = value + 1.0

    vector.canoe.tfs.Report.test_step("", f"Setting TestVariableWriteable to {value}")
    XILServer.MAPort.TestVariableWriteable = value

    vector.canoe.threading.wait_for_timeout(200)
    
    vector.canoe.tfs.Report.test_step("", f"Checking TestVariableReadable value. Expected: {expectedValue}")
    if(XILServer.MAPort.TestVariableReadable.copy() == expectedValue):
      vector.canoe.tfs.Report.test_step_pass("", "Test passed!")
      return
    
    vector.canoe.tfs.Report.test_step("", f"Actual: {XILServer.MAPort.TestVariableReadable.copy()}")
    vector.canoe.tfs.Report.test_step_fail("", "Test failed!")
  